packages/browser-wallet/src/shared/utils/token-helpers.ts

```diff
@@ -399,16 +402,25 @@ export const getTokens = async (
         return [metadataUrls, metadata];
     })();
 
-    const balancesPromise = getContractBalances(client, contractDetails.index, contractDetails.subindex, ids, account);
+    const balancesPromise = getContractBalances(
+        client,
+        contractDetails.index,
+        contractDetails.subindex,
+        ids,
+        account,
+        onError
+    );
 
     const [[metadataUrls, metadata], balances] = await Promise.all([metadataPromise, balancesPromise]); // Run in parallel.
 
-    return ids.map((id, i) => ({
-        id,
-        metadataLink: metadataUrls[i],
-        metadata: metadata[i],
-        balance: balances[id] ?? 0n,
-    }));
+    return ids
+        .filter((id, i) => id in balances && metadata[i])
+        .map((id, i) => ({
+            id,
```

Comment:
I realise that this needs to be rewritten,so the indices do not get shifted. (Probably gonna merge into a reduce)

Reply:
Is it okay @soerenbf?